pkg/storage/replica.go

```diff
@@ -2749,7 +2793,17 @@ func (r *Replica) handleRaftReadyRaftMuLocked(
 	r.store.raftEntryCache.addEntries(r.RangeID, rd.Entries)
 	r.mu.lastIndex = lastIndex
 	r.mu.raftLogSize = raftLogSize
-	r.mu.leaderID = leaderID
+	if r.mu.leaderID != leaderID {
+		r.mu.leaderID = leaderID
+		if r.mu.leaderID == r.mu.replicaID {
+			r.mu.proposalQuotaBase = r.mu.lastIndex
+			r.proposalQuota.reset(leaderProposalQuota)
+		} else {
+			r.proposalQuota.reset(followerProposalQuota)
```

Comment:
So they can propose requests which get forwarded to the leader. For example, followers propose lease requests which is how they determine who holds the lease. We could probably change this so that we don't acquire proposal quota when a follower proposes a Raft command.